src/util/worker/mod.rs

```diff
@@ -367,4 +507,37 @@ mod test {
         }
         assert_eq!(rx.recv().unwrap(), 0);
     }
+
+    #[test]
+    fn test_on_tick() {
+        let mut worker = Builder::new("test-worker-tick")
+            .batch_size(4)
+            .tick_timeout(Duration::from_secs(3))
+            .tasks_per_tick(5)
+            .create();
+        for _ in 0..20 {
+            worker.schedule("normal msg").unwrap();
+        }
+        thread::sleep(Duration::from_secs(3));
+
+        let (tx, rx) = mpsc::channel();
+        worker.start(TickRunner { ch: tx }).unwrap();
+
+        // The stream in rx should be: ^^^^^^^^o^^^^^^^^o^^^^oo.
+        // '^' means normal message and 'o' means tick message.
+        for i in 0..22 {
+            let msg = rx.recv_timeout(Duration::from_secs(3)).unwrap();
+            if i != 8 && i != 17 {
```

Comment:
why can we guarantee 8 and 17 is for tick?

Reply:
Because we produce `normal msg` quickly, so we won't meet `timeout`. And `tasks_per_tick` is `5` and `batch_size` is `4`, so we will yield a `tick msg` after we finish 2 batch, which is 8 tasks.